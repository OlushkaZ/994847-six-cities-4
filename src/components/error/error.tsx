import * as React from 'react';
import {connect} from "react-redux";

interface Props {
  show: boolean;
}

const Error: React.FC<Props> = ({show}) => {
  if (!show) {
    return null;
  }

  return (
    <div
      style={{
        position: `fixed`,
        top: 0,
        right: `50%`,
        width: `300px`,
        border: `2px solid red`,
        backgroundColor: `white`,
        fontSize: `20px`,
        fontWeight: `bold`,
        textAlign: `center`,
      }}>Что-то пошло не так</div>
  );
};

const mapStateToProps = (state) => ({
  show: state.data.isError
});

export {Error};
export default connect(mapStateToProps, null)(Error);

