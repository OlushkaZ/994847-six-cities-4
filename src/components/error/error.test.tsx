import React from 'react';
import renderer from 'react-test-renderer';

import {Error} from './error';

it(`Should Error render correctly`, () => {
  const tree = renderer
    .create(
        <Error show={true}/>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
