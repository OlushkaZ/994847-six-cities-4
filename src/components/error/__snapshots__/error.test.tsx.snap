// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Error render correctly 1`] = `
<div
  style={
    Object {
      "backgroundColor": "white",
      "border": "2px solid red",
      "fontSize": "20px",
      "fontWeight": "bold",
      "position": "fixed",
      "right": "50%",
      "textAlign": "center",
      "top": 0,
      "width": "300px",
    }
  }
>
  Что-то пошло не так
</div>
`;
